core/new-gui/src/app/workspace/component/codearea-custom-template/codearea-custom-template.component.ts

```diff
@@ -20,20 +20,40 @@ import { CoeditorPresenceService } from "../../service/workflow-graph/model/coed
   templateUrl: "./codearea-custom-template.component.html",
   styleUrls: ["./codearea-custom-template.component.scss"],
 })
-export class CodeareaCustomTemplateComponent extends FieldType {
+export class CodeareaCustomTemplateComponent extends FieldType<any> implements AfterViewInit {
   dialogRef: MatDialogRef<CodeEditorDialogComponent> | undefined;
+  readonly: boolean = false;
 
   constructor(public dialog: MatDialog, private coeditorPresenceService: CoeditorPresenceService) {
     super();
     this.handleShadowingMode();
   }
 
+  ngAfterViewInit() {
+    this.handleReadonlyStatusChange();
+  }
+
+  /**
+   * Syncs the disabled status of the button with formControl.
+   * Used to fit the unit test since undefined might occur.
+   * TODO: Using <code>formControl</code> here instead of
+   *  <code>WorkflowActionService.checkWorkflowModificationEnabled()</code>
+   *  since the readonly status of operator properties also locally depend on whether "Unlock for Logic Change"
+   *  is enabled, which can only be accessed via <code>formControl</code>. Might need a more unified solution.
+   */
+  handleReadonlyStatusChange(): void {
+    if (this.field !== undefined)
```

Comment:
It's Better to check for undefined and return early explicitly. Please swap the of condition.

Reply:
Done.